localenv/local-http-signatures/postman-scripts/preRequestSignatures.js

```diff
@@ -1,32 +1,42 @@
-const requestUrl = request.url
-  .replace(/{{([A-Za-z]\w+)}}/g, (_, key) => pm.collectionVariables.get(key))
+let requestUrl = request.url
+  .replace(/{{([A-Za-z]\w+)}}/g, (_, key) => pm.environment.get(key))
   .replace(/localhost:([3,4])000/g, (_, key) =>
-    key === '3' ? 'cloud-nine-wallet-backend' : 'happy-life-bank-backend'
+    key === '3'
+      ? pm.environment.get('host3000')
+      : pm.environment.get('host4000')
   )
+
 const requestBody =
   request.method === 'POST' && Object.keys(request.data).length !== 0
-    ? request.data.replace(/{{([A-Za-z]\w+)}}/g, (_, key) =>
-        pm.collectionVariables.get(key)
-      )
+    ? request.data
+        .replace(/{{([A-Za-z]\w+)}}/g, (_, key) => pm.environment.get(key))
+        .replace(/http:\/\/localhost:([3,4])000/g, (_, key) =>
+          key === '3'
+            ? 'https://' + pm.environment.get('host3000')
+            : 'https://' + pm.environment.get('host4000')
+        )
     : undefined
+pm.request.body.raw = requestBody
+
 const requestHeaders = JSON.parse(
   JSON.stringify(request.headers).replace(/{{([A-Za-z]\w+)}}/g, (_, key) =>
-    pm.collectionVariables.get(key)
+    pm.environment.get(key)
   )
 )
 
 // Request Signature Headers
 pm.sendRequest(
   {
-    url: pm.collectionVariables.get('signatureUrl'),
+    url: pm.environment.get('signatureUrl'),
     method: 'POST',
     header: {
       'content-type': 'application/json'
     },
     body: {
       mode: 'raw',
       raw: JSON.stringify({
-        keyId: pm.collectionVariables.get('keyId'),
+        keyId: pm.environment.get('clientKeyId'),
+        base64Key: pm.environment.get('clientPrivateKey'),
```

Comment:
should we specify the variable name explicitly `clientBase64PrivateKey`? Or can we provide the private key directly and encode it before the request?

Should definitely have the option in test net to copy both, the normal and the base64 keys potentially